service/src/main/java/bio/terra/workspace/service/danglingresource/DanglingResourceCleanupService.java

```diff
@@ -0,0 +1,235 @@
+package bio.terra.workspace.service.danglingresource;
+
+import bio.terra.common.logging.LoggingUtils;
+import bio.terra.workspace.app.configuration.external.DanglingResourceCleanupConfiguration;
+import bio.terra.workspace.db.CronjobDao;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.resource.controlled.ControlledResourceService;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.ainotebook.ControlledAiNotebookInstanceResource;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.dataproccluster.ControlledDataprocClusterResource;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.gceinstance.ControlledGceInstanceResource;
+import bio.terra.workspace.service.resource.controlled.flight.delete.DeleteControlledResourcesFlight;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.model.WsmResource;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.OperationType;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DanglingResourceCleanupService {
+  private static final Logger logger =
+      LoggerFactory.getLogger(DanglingResourceCleanupService.class);
+  private static final String DANGLING_RESOURCE_CLEANUP_JOB_NAME = "dangling_resource_cleanup_job";
+
+  private final DanglingResourceCleanupConfiguration configuration;
+  private final ControlledResourceService controlledResourceService;
+  private final JobService jobService;
+  private final ResourceDao resourceDao;
+  private final CronjobDao cronjobDao;
+  private final SamService samService;
+  private final CrlService crlService;
+  private final ScheduledExecutorService scheduler;
+
+  private final List<WsmResourceType> DANGLING_RESOURCE_TYPES =
+      List.of(
+          WsmResourceType.CONTROLLED_GCP_AI_NOTEBOOK_INSTANCE,
+          WsmResourceType.CONTROLLED_GCP_GCE_INSTANCE,
```

Comment:
Is it currently possible for notebooks or GCE instances to be deleted on the cloud side yet? My understanding was this was dataproc only atm

Reply:
Not currently, should I remove notebooks and gce instances from the list here?